File: learn the basics/maths/divisors.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

// Function to find all divisors of a number
vector<int> divisors(int n) {
    vector<int> divisors_list;
    int sqrtn = sqrt(n);
    for (int i = 1; i <= sqrtn; ++i) {
        if (n % i == 0) {
            divisors_list.push_back(i);
            if (i != n / i) {
                divisors_list.push_back(n / i);
            }
        }
    }
    return divisors_list;
}

// Function to calculate ΣF(i) for i from 1 to n
long long divisorstilln(int n) {
    long long total_sum = 0;
    for (int i = 1; i <= n; ++i) {
        total_sum += i * (n / i);
    }
    return total_sum;
}

int main() {
    int n;
    cout << "Enter a number: ";
    cin >> n;

    long long result = divisorstilln(n);
    cout << "The sum of ΣF(i) from 1 to " << n << " is: " << result << endl;

    return 0;
}

File: learn the basics/maths/divisors.py
import math
def divisor(n):
    divisor_list = []
    for i in range(1,math.sqrt(n)):
        if n%i==0:
            divisor_list.append(i)
            if i != n//i:
                divisor_list.append(n//i)
    return divisor_list
# Function to calculate ΣF(i) for i from 1 to n
def prob(n):
    total_sum = 0
    for i in range(1,n):
        total_sum += i * (n//i)
    return total_sum